Force 29 and Z to be haploid, do variant calling and then can identify which snps are Z and which are W
Should see a lower fraction of reads with the W variant aligning to the Z
But when I do haploid calling, this should get rid of most of the false positives, but expect more false positives on 29 than on Z
Remake plot with total number of snps in females vs males

On Z use all snps on males and then parse snps on female Z into those shared with males and those that are not
Then look at AF of these subgroups and we'd expect... 
SNPs shared with males would have 

Add in an autosome (medium sized, similar to Z size) and look at haploid calling SNP density means of males and females and 10kb scans
Identify the PAR boundary:

1-2 million is no longer recombining in females and is likely where the fusion occured

Gene analysis: look for genes within windows that are 